python/flask/open-redirect/open-redirect.yml

```diff
@@ -0,0 +1,59 @@
+rules:
+  - id: open-redirect
+    patterns:
+      - pattern-inside: |
+          @$APP.route(...)
+          def $X():
+            ...
+      - pattern-either:
+        - pattern: |
+            $V = flask.request.$W.get($X)
+            ...
+            flask.redirect($V)
+        - pattern: |
+            $V = flask.request.$W.get($X)
+            ...
+            $Z = flask.redirect($V)
+        - pattern: |
+            $V = flask.request.$W.get($X)
+            ...
+            return flask.redirect($V)
+        - pattern: |
+            $V = flask.request.$W[$X]
+            ...
+            flask.redirect($V)
+        - pattern: |
+            $V = flask.request.$W[$X]
+            ...
+            $Z = flask.redirect($V)
+        - pattern: |
+            $V = flask.request.$W[$X]
+            ...
+            return flask.redirect($V)
+        - pattern: |
+            $V = flask.request.$W($X)
+            ...
+            flask.redirect($V)
+        - pattern: |
+            $V = flask.request.$W($X)
+            ...
+            $Z = flask.redirect($V)
+        - pattern: |
+            $V = flask.request.$W($X)
+            ...
+            return flask.redirect($V)
+        - pattern: |
+            $V = flask.request.$W
+            ...
+            flask.redirect($V)
+        - pattern: |
+            $V = flask.request.$W
+            ...
+            $Z = flask.redirect($V)
+        - pattern: |
+            $V = flask.request.$W
+            ...
+            return flask.redirect($V)
+    message: "Found data from flask.request passed in to flask.redirect. This is an open redirect and could be exploited"
```

Comment:
It took me second to decypher the flask.request passed in to flask.redirect sentence. Does it make sense to just say:
```
Data from request is passed to redirect(). This is an open redirect and could be exploited. See <LINK>
```

Reply:
That's much better! Thanks!